files/dx_sync_directory.py

```diff
@@ -569,15 +574,22 @@ def main():
 
     tars_to_upload = split_into_tar_files(files_to_upload, log, args)
 
-    for tar in tars_to_upload:
-        log = create_tar_file(tar, log, args)
-        log = upload_tar_files(log, args)
-        log = remove_tar_files(log, args)
-
     # Run through upload & remove in case last invocation was interrupted
     if len(tars_to_upload) == 0:
-        log = upload_tar_files(log, args)
+        log, _ = upload_tar_files(log, args)
+        log = remove_tar_files(log, args)
+
+    for tar in tars_to_upload:
+        start_ts = time.time()
```

Comment:
start time should be before any files are uploaded.
assuming 15 min intervals
scenarion 1
tar file upload starts at 1:05,
calculate the duration for the first upload, just say it takes 22m
assume next tar file will also take 22m
if hour(duration + current time) > hour(current time), then exit (because it is going to finish after the next cron run)